### ISSUE TITLE
Very high cls loss when training on custom dataset

### ISSUE THREAD

**Issue by mast1ren** [AILab-CVC/YOLO-World#431]
I met a strange problem. When I change the pretrain weight to `yolo_world_l_clip_base_dual_vlpan_2e-3adamw_32xb16_100e_0365_goldg_train_pretrained-0e566235` and keep other config on my own data, I got a very high loss_cls and other losses are normal.

`Epoch(train)    [1][  50/1188]  base_lr: 1.0000e-05 lr: 1.3749e-07 eta: 1 day, 11:56:13   time: 1.0895    data_time:0.0323    memory: 9558   grad_norm: nan    loss: 5224555.7200    loss_cls: 5224534.3800    loss_bbox: 11.9264    loss_dfl: 9.4203`

It is normal when using pretrain weight `yolo_world_v2_l_obj365v1_goldg_pretrain_1280ft-9babe3f6.pth` 

here is the config:
```
_base_ = ('../../third_party/mmyolo/configs/yolov8/'
    'yolov8_l_syncbn_fast_8xb16-500e_coco.py')
custom_imports = dict(
    imports=['yolo_world'],
    allow_failed_imports=False)

# hyper-parameters
num_classes = 200
num_training_classes = 148
max_epochs = 100  # Maximum training epochs
close_mosaic_epochs = 10
save_epoch_intervals = 5
text_channels = 512
neck_embed_channels = [128, 256, _base_.last_stage_out_channels // 2]
neck_num_heads = [4, 8, _base_.last_stage_out_channels // 2 // 32]
base_lr = 1e-5
weight_decay = 0.05
train_batch_size_per_gpu = 2
load_from = 'pretrained_models/yolo_world_v2_l_obj365v1_goldg_cc3mlite_pretrain-ca93cd1f.pth'
# load_from = 'pretrained_models/yolo_world_l_clip_base_dual_vlpan_2e-3adamw_32xb16_100e_o365_goldg_train_pretrained-0e566235.pth'
text_model_name = 'pretrained_models/clip-vit-base-patch32'
persistent_workers = False
img_scale = (1280, 1280)

# model settings
model = dict(
    type='YOLOWorldDetector',
    mm_neck=True,
    num_train_classes=num_training_classes,
    num_test_classes=num_classes,
    data_preprocessor=dict(type='YOLOWDetDataPreprocessor'),
    backbone=dict(
        _delete_=True,
        type='MultiModalYOLOBackbone',
        image_model={{_base_.model.backbone}},
        text_model=dict(
            type='HuggingCLIPLanguageBackbone',
            model_name=text_model_name,
            frozen_modules=[''])),
    neck=dict(type='YOLOWorldDualPAFPN',
              guide_channels=text_channels,
              embed_channels=neck_embed_channels,
              num_heads=neck_num_heads,
              block_cfg=dict(type='MaxSigmoidCSPLayerWithTwoConv'),
              text_enhancder=dict(type='ImagePoolingAttentionModule',
                                  embed_channels=256,
                                  num_heads=8)),
    bbox_head=dict(type='YOLOWorldHead',
                   head_module=dict(type='YOLOWorldHeadModule',
                                    use_bn_head=True,
                                    embed_dims=text_channels,
                                    num_classes=num_training_classes)),
    train_cfg=dict(assigner=dict(num_classes=num_training_classes)))

# dataset settings
text_transform = [
    dict(type='RandomLoadText',
         num_neg_samples=(num_classes, num_classes),
         max_num_samples=num_training_classes,
         padding_to_max=True,
         padding_value=''),
    dict(type='mmdet.PackDetInputs',
         meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape', 'flip',
                    'flip_direction', 'texts'))
]
mosaic_affine_transform = [
    dict(
        type='MultiModalMosaic',
        img_scale=_base_.img_scale,
        pad_val=114.0,
        pre_transform=_base_.pre_transform),
    dict(
        type='YOLOv5RandomAffine',
        max_rotate_degree=0.0,
        max_shear_degree=0.0,
        max_aspect_ratio=100.,
        scaling_ratio_range=(1 - _base_.affine_scale,
                             1 + _base_.affine_scale),
        border=(-_base_.img_scale[0] // 2, -_base_.img_scale[1] // 2),
        border_val=(114, 114, 114),
        )
]
train_pipeline = [
    *_base_.pre_transform,
    dict(type='MultiModalMosaic',
         img_scale=_base_.img_scale,
         pad_val=114.0,
         pre_transform=_base_.pre_transform),
    dict(
        type='YOLOv5RandomAffine',
        max_rotate_degree=0.0,
        max_shear_degree=0.0,
        scaling_ratio_range=(1 - _base_.affine_scale, 1 + _base_.affine_scale),
        max_aspect_ratio=_base_.max_aspect_ratio,
        border=(-_base_.img_scale[0] // 2, -_base_.img_scale[1] // 2),
        border_val=(114, 114, 114)),
    *_base_.last_transform[:-1],
    *text_transform,
]
train_pipeline_stage2 = [
    *_base_.train_pipeline_stage2[:-1],
    *text_transform
]

v1_ovd_train_dataset = dict(
    type='YOLOv5MixedGroundingDataset',
    data_root='/data/V1/',
    ann_file='annotations/annotations_en_all_ovd.json',
    data_prefix=dict(img='all/'),
    filter_cfg=dict(filter_empty_gt=False, min_size=32),
    pipeline=train_pipeline
)

train_dataloader = dict(
    persistent_workers=persistent_workers,
    batch_size=train_batch_size_per_gpu,
    collate_fn=dict(type='yolow_collate'),
    dataset=v1_ovd_train_dataset)
test_pipeline = [
    *_base_.test_pipeline[:-1],
    dict(type='LoadText'),
    dict(
        type='mmdet.PackDetInputs',
        meta_keys=('img_id', 'img_path', 'ori_shape', 'img_shape',
                   'scale_factor', 'pad_param', 'texts'))
]

# training settings
default_hooks = dict(
    param_scheduler=dict(
        scheduler_type='linear',
        lr_factor=0.01,
        max_epochs=max_epochs),
    checkpoint=dict(
        max_keep_ckpts=-1,
        save_best=None,
        interval=save_epoch_intervals,
        rule='greater'))
custom_hooks = [
    dict(
        type='EMAHook',
        ema_type='ExpMomentumEMA',
        momentum=0.0001,
        update_buffers=True,
        strict_load=False,
        priority=49),
    dict(
        type='mmdet.PipelineSwitchHook',
        switch_epoch=max_epochs - close_mosaic_epochs,
        switch_pipeline=train_pipeline_stage2)
]
train_cfg = dict(
    max_epochs=max_epochs,
    val_interval=2,
    dynamic_intervals=[((max_epochs - close_mosaic_epochs),
                        _base_.val_interval_stage2)])
optim_wrapper = dict(optimizer=dict(
    _delete_=True,
    type='SGD',
    lr=base_lr,
    momentum=0.937,
    nesterov=True,
    weight_decay=weight_decay,
    batch_size_per_gpu=train_batch_size_per_gpu),
                     paramwise_cfg=dict(
                         custom_keys={
                             'backbone.text_model': dict(lr_mult=0.01),
                             'logit_scale': dict(weight_decay=0.0)
                         }),
                     constructor='YOLOWv5OptimizerConstructor')
```

**Comment by mast1ren** [comment]
problem solved, `use_bn_head` in model head config should be set to `False` when using v1 weight

**Comment by wenqiuL** [comment]
@mast1ren 你好，我也出现了同样的问题，请问这个v1 weight是因为load_from = 'pretrained_models/yolo_world_v2_l_obj365v1_goldg_cc3mlite_pretrain-ca93cd1f.pth'吗？另外以上你提供的代码文件是哪个config文件呢？

**Comment by mast1ren** [comment]
> @mast1ren 你好，我也出现了同样的问题，请问这个v1 weight是因为load_from = 'pretrained_models/yolo_world_v2_l_obj365v1_goldg_cc3mlite_pretrain-ca93cd1f.pth'吗？另外以上你提供的代码文件是哪个config文件呢？

load_from = 'pretrained_models/yolo_world_v2_l_obj365v1_goldg_cc3mlite_pretrain-ca93cd1f.pth'是使用的v2的weight，我使用的config也是v2的，在这个v2的weight上面没有出现loss问题。如果你使用v1的weight那你需要对config做一些更改。

V1：https://github.com/AILab-CVC/YOLO-World/blob/b4fd87838d7f53adc0dbf5844313b92d9e3124c7/configs/pretrain_v1/yolo_world_l_dual_vlpan_l2norm_2e-3_100e_4x8gpus_obj365v1_goldg_train_lvis_minival.py#L42-L45
V2：https://github.com/AILab-CVC/YOLO-World/blob/b4fd87838d7f53adc0dbf5844313b92d9e3124c7/configs/pretrain/yolo_world_v2_l_clip_large_vlpan_bn_2e-3_100e_4x8gpus_obj365v1_goldg_train_800ft_lvis_minival.py#L42-L46

![图片](https://github.com/user-attachments/assets/b9b1afaa-a114-4383-8374-2edaab706f3f)

**Comment by wenqiuL** [comment]
@mast1ren 好的，非常感谢您的回复~
